lib/config.js

```diff
@@ -184,24 +51,31 @@ function getLocalConfig(thisConfig, directory) {
 class Config {
 
     /**
-     * Config options
      * @param {Object} options Options to be passed in
      */
     constructor(options) {
         options = options || {};
 
+        this.options = options;
         this.ignore = options.ignore;
         this.ignorePath = options.ignorePath;
-        this.cache = {};
         this.parser = options.parser;
         this.parserOptions = options.parserOptions || {};
 
-        this.baseConfig = options.baseConfig ? loadConfig(options.baseConfig) : { rules: {} };
+        this.baseConfig = options.baseConfig
+            ? ConfigOps.merge({}, ConfigFile.loadObject(options.baseConfig))
+            : { rules: {} };
+        this.baseConfig.filePath = "";
+        this.baseConfig.baseDirectory = this.options.cwd;
+
+        configCache.reset();
```

Comment:
Refactored to create a new `ConfigCache` in the `Config` constructor rather than using a singleton.